### PR TITLE
First round of feedback

### PR DESCRIPTION
**Pull request by brianegan**
Hey Salih, hope all is well :) Great workshop! I've been asked by the Flutter team to take each workshop from the GDEs and provide feedback. The easiest way I've found is via pull requests. 

~~I've left some feedback inline, and general feedback in a file called `feedback.md`.~~ Feedback is now below! I'm in no way an authority figure demanding changes, this feedback is only meant to be helpful. Please feel free to accept the feedback that makes sense to you and ignore the rest :)

  - The app in step 1 is awesome. Looks great and makes me excited to go through
    the workshop!
  - I've done some random copy editing here and there. I don't mean to remove
    your tone of voice, just improve text I found a little awkward or hard to
    understand. Please feel free to accept or reject any text edits :)  
  - I wish each step had a title / heading to ground me and provide context
    about what this step is about. What is the main topic of each step? What
    should I focus on?
  - Hm, Getting a "script error" when I run Step 2, both with my solution and
    the solution provided by the workshop. Perhaps it cannot request focus as a
    separate panel?
  - Sometimes I didn't see the "Homework" sections. Ironically, the blockquote
    styling kind of de-emphasized those sections for me visually, especially
    when placed underneath a code block. Emoji's & Headings may help the user.
  - In some cases, I felt like code comments might be a bit easier to read than
    a block of code a block of text underneath. Just my style though, again
    please feel free to ignore if you like it the other way :)
  - I like how the early steps sort of guide the reader through how the pieces
    fit together and ignore implementation a bit, then later steps ask the user
    to implement those topics with a knowledge of how the pieces fit together. I
    was worried at first there was a bit too much copy/paste happening, but it
    made sense in the end!
  - It's easy to follow where code should be placed. 
  - Overall, I really enjoyed the content of the workshop! It combined not only
    the keyboard system but it was fun to think about the game logic as well.
    Nice one :)

### CODE REVIEW

**Comment by salihgueler** [comment]
Thank you for your feedback Brian! I will merge this in and address everything else with a seperate commit. I will leave the commit reference here in case you want to follow up.

File: step_04/solution.dart [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
 
+// Hmmm, is this solution working for you in DartPad? I worry I might be running
+// into a bug :/ I cannot use Escape to clear text, and when I hit Control +
+// Enter it simply moves to the next page?
 class LoginPage extends StatefulWidget {
```
Comment (salihgueler) [inline step_04/solution.dart:7]: I tested it now as well and it is working, but maybe the cache/focus problem that is discussed before is happening :(

Comment (brianegan) [reply step_04/solution.dart:7]: Yah it seems inconsistent. Luckily today everything is going smooth so far!

Comment (salihgueler) [reply step_04/solution.dart:7]: I made couple of tiny improvements to increase the user experience :)

**Pull request merged**